numpy/_core/src/multiarray/item_selection.c

```diff
@@ -2138,26 +2137,20 @@ PyArray_SearchSorted(PyArrayObject *op1, PyObject *op2,
     }
     if (binsearch == NULL && argbinsearch == NULL) {
         PyErr_SetString(PyExc_TypeError, "compare not supported for type");
-        /* refs to dtype we own = 1 */
         Py_DECREF(dtype);
-        /* refs to dtype we own = 0 */
         return NULL;
     }
 
-    /* need ap2 as contiguous array and of right type */
-    /* refs to dtype we own = 1 */
-    Py_INCREF(dtype);
-    /* refs to dtype we own = 2 */
+    /* need ap2 as contiguous array and of right dtype (steals and may be replace it) */
```

Comment:
`replace it` -> `replace dtype` (since "it" could also refer to `ap2`, the array).

Actually, with my comment below, I think this can just be
```
/* need ap2 as contiguous array and of right dtype (note: steals dtype reference) */